### ISSUE TITLE
[feature] Add wind indicators to AIS targets (AIS+) when available

### ISSUE THREAD

**Issue by zubenubi** [SignalK/freeboard-sk#63]
At a scale that allows for the chart display to encompass up to 3-4nm, add wind indicator/heading/course lines to the sk-enabled AIS targets that provide such data.

Exchanging data between between vessels is enabled by signalk and plugins. This means that sailing clubs can streamline their operations on the water by automating the exchange of wind sensor data. With this data displayed on a chart for race committee officials, accuracy is improved and radio communications traffic is reduced.

freeboard-sk already displays AIS targets' position and vessel name/mmsi. This new feature would add wind indicator lines similar to those seen on the 'self' vessel for the ais targets that have the data available. Considering that the density of ais traffic is of course highly variable at this scale, it might be preferable to display these sk-enabled 'AIS+' targets in a different color scheme (orange or green shades?) or have an option to display them exclusively apart from standard ais targets.

**Comment by zubenubi** [comment]
Yes. Enthusiastically! The use-case is most interested in a set of vessels working within 5nm of each other. Given the possible clutter of AIS targets it would be awesome to have the ability to filter to only display the AIS targets of interest and to augment those with wind vectors. I assume that the filtering mechanism would be able to select vessels by name, mmsi, uuid, or (perhaps easier) by whether or not those vessels have wind data?

**Comment by panaaj** [comment]
@zubenubi I am in the process of including the wind vector on AIS targets, is it True or Apparent wind that should be displayed

**Comment by rob42** [comment]
I think true wind is more useful. Apparent wind on an ais target would only be useful to the targets crew.

**Comment by zubenubi** [comment]
Apparent Wind is probably more ubiquitous and less ambiguous. True wind has a number of formulations (True Wind over water {STW, set/drift derived}, True wind over ground {GPS/SOG-cog derived}). Im sure we are all familiar. An exhaustive discussion is on Panbo:
https://mt.panbo.com/2010/11/what-wind-instrument.html

A graphic representation:
https://cdnkey-bb0b.kxcdn.com/assets_c/2009/11/WindVectors2d5-1261.html

At a recent event the discussion was heated over what True Wind really is. The conclusion in that discussion is that the True Wind preferred by small boat sailors for aligning a course is the True Wind over water incorporating STW and set/drift. So to measure this an RC official would not be underway nor at anchor but *drifting* while making those measurements - which is effectively 'Apparent Wind when the boat vrom which is is being measured is drifting.

So I am wondering if it is possible for the end user to be able to select the signalk path they would prefer to use in the calculation of a wind vector displayed on an AIS target?

**Comment by rob42** [comment]
The problem with apparent wind is that it changes with vessel speed and direction. We sail a cat and apparent wind can be 30-50deg ahead of true wind. If you saw that on a chartplotter you would assume that the wind there was quite different from the true direction.

Even though there are many true wind definitions, they are all quite close in direction, eg < 5deg in most cases, and the inaccuracies in measurement and display rendering will make that difficult to see anyway.

It depends on what you are looking for. For me its some insight into wind in the vicinity of the boat, so I can plan tacks etc. Cant get that from apparent without lots of maths.

**Comment by panaaj** [comment]
Given the potential varying needs I will make it a configurable option in `Settings` to select whether the `True Wind` or `Apparent Wind` value is used to display the AIS wind vector.

**Comment by tkurki** [comment]
I think the word AIS is misleading here, as showing wind information for other vessels has nothing to do with AIS.

If you can get wind data over SK you can position, speed etc over SK, with no AIS in the receiving or sharing vessel. Conversely having AIS transmitter or receiver will not enable wind data in any way.

So I suggest changing the issue title and description to avoid confusion.
